src/main/java/com/example/auctionapp/repository/ItemRepository.java

```diff
@@ -1,10 +1,17 @@
 package com.example.auctionapp.repository;
 
 import com.example.auctionapp.model.Item;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 import org.springframework.stereotype.Repository;
 
 @Repository
 public interface ItemRepository extends JpaRepository<Item, Long> {
 
+    @Query("select i from Item i where i.categoryId = :categoryId")
+    Page<Item> getItemsByCategoryId(@Param("categoryId") long categoryId, Pageable pageable);
```

Comment:
Do we need custom query here, I believe you could use existing 
`Page<Item> findByCategoryId(Long categoryId, Pageable pageable);`